src/utils.ts

```diff
@@ -39,3 +40,12 @@ export function findHeadingBoundary(
 
   return null;
 }
+
+
+export async function createDirNotExist(filepath: string) {
+  try {
+    await this.app.vault.createFolder(path.dirname(filepath))
```

Comment:
Reading through the comment you noted, @SettingDust (from https://discord.com/channels/686053708261228577/840286264964022302/1104023861449728060; copied here for clarity):

> You can run the code in  the developer console (Ctrl+Shift+I).
> 
> Testing in v1.2.7 shows that this.app.vault.adapter.mkdir can indeed create multiple path segments, even if the folder already exists:
> await this.app.vault.adapter.mkdir("a/b/c")
> 
> However, you should prefer the vault API (https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+the+Vault+API+over+the+Adapter+API), so you should use this.app.vault.createFolder. However, while this can also create multiple path segments, it does not like it if the folder already exists. So you could write:
> 
> ```
> if (!this.app.vault.getAbstractFileByPath("a/b/c")) {
>   await this.app.vault.createFolder("a/b/c")
> }
> ```
> 
> Alternatively, catch the exception thrown by it if the folder already exists (EAFP):
> 
> ```
> try {
>   await this.app.vault.createFolder("a/b/c")
> } catch {
>   // the folder already exists, but we don't care
> }
> ```

It sounds like this might fail if _any_ path segments already exist -- so if, for example, we were trying to write a file under `a/b/c/filename.md` and `a/b/` already existed, but `c/` did not, his phrasing in the comment you linked suggests that this *might* throw an error.  Can you verify that it doesn't?  If it does, it seems like that's something we should fix, and, if so, I think my earlier PR review comments might give you some tips.

Reply:
Tried. I created `a` folder. Execute put `vault/a/b/c/d.md`. No error.